src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingAssessmentResource.java

```diff
@@ -144,7 +144,19 @@ public ProgrammingAssessmentResource(AuthorizationCheckService authCheckService,
             return forbidden();
         }
 
-        if (newResult.getScore() != null && newResult.getScore() < 100 && newResult.isSuccessful()) {
+        if (Boolean.FALSE.equals(newResult.isRated())) {
+            throw new BadRequestAlertException("Result is not rated", ENTITY_NAME, "resultNotRated");
+        }
+        if (newResult.getResultString() == null) {
+            throw new BadRequestAlertException("Result string is required.", ENTITY_NAME, "resultStringNull");
+        }
+        else if (newResult.getResultString().length() > 255) {
+            throw new BadRequestAlertException("Result string is too long.", ENTITY_NAME, "resultStringNull");
+        }
+        else if (newResult.getScore() == null) {
+            throw new BadRequestAlertException("Score is required.", ENTITY_NAME, "scoreNull");
+        }
+        else if (newResult.getScore() < 100 && newResult.isSuccessful()) {
```

Comment:
I'd suggest moving the validation of `newResult` as far up as possible in the method. Validation should be done before services are used.

Reply:
The calls which are done before are also checks, e.g. if the user has enough roles etc. I would leave it like it is, as I think it is equally important to check if he has the rights to submit/save a result or that the result has the correct format. :)